vitals/BatteryLimit.java

```diff
@@ -0,0 +1,74 @@
+package vitals;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class BatteryLimit {
+
+	public static String measureUnitDefault = "Celsius";
+	static List<String> unitsTemperature = Arrays.asList("Celsius", "Fahrenheit");
+
+	BatteryCheck bmsCheck;
+
+	public BatteryLimit(BatteryCheck batteryMeasures) {
+		this.bmsCheck = batteryMeasures;
+	}
+
+	public BatteryLimit() {
+		// constructor stub
+	}
+
+	public boolean checkBatteryOk() {
+		boolean check_of_charge = isChargeWithinLimit(bmsCheck.ChargeRate);
+		boolean check_of_temp = isTemperatureWithinLimit(bmsCheck.Temperature);
+		boolean check_of_soc = isSocWithinLimit(bmsCheck.StateOfCharge);
+		return ((check_of_temp && check_of_soc) && check_of_charge);
+	}
+
+	public boolean isTemperatureWithinLimit(float temperature) {
+
+		if ((measureUnitDefault).equals(unitsTemperature.get(1))) {

```

Comment:
looks like compromised condition check ....it always checks with specific index based item

Reply:
Logic is changed